logstash-core/lib/logstash/environment.rb

```diff
@@ -46,7 +46,11 @@ module Environment
 
   ].each {|setting| SETTINGS.register(setting) }
 
-  SETTINGS.register Setting::WritableDirectory.new("path.queue", ::File.join(SETTINGS.get("path.data"), "queue"))
+  queue_file_path = ::File.join(SETTINGS.get("path.data"), "queue")
+  # Here we have to create the directory even though it might be overridden by settings.yml
+  # We should consider a Setting::LazyDirectory class that creates the directory on first 'get'
+  FileUtils.mkdir_p([queue_file_path])
```

Comment:
I am wondering if this is the right place to create the dir if it does not exists? see https://github.com/elastic/logstash/blob/263e973d090a28446b64a81fa812e1109620f790/rakelib/test.rake#L10-L13  where I did this tmp patch to make sure the dir also exists for the tests ... /cc @guyboertje